Core/src/org/sleuthkit/autopsy/centralrepository/datamodel/CorrelationAttributeUtil.java

```diff
@@ -137,54 +139,56 @@ private static String getEmailAddressAttrDisplayName() {
                     }
-                }
-                Content parent = analysisResult.getParent();
-                if (parent instanceof AbstractFile) {
-                    correlationAttrs.addAll(CorrelationAttributeUtil.makeCorrAttrsForSearch((AbstractFile) parent));
-                } else if (parent instanceof AnalysisResult) {
-                    correlationAttrs.addAll(CorrelationAttributeUtil.makeCorrAttrsForSearch((AnalysisResult) parent));
-                } else if (parent instanceof DataArtifact) {
-                    correlationAttrs.addAll(CorrelationAttributeUtil.makeCorrAttrsForSearch((DataArtifact) parent));
-                } else if (parent instanceof OsAccount) {
-                    for (OsAccountInstance osAccountInst : ((OsAccount) parent).getOsAccountInstances()) {
-                        if (osAccountInst.getDataSource().equals(analysisResult.getDataSource())) {
-                            correlationAttrs.addAll(CorrelationAttributeUtil.makeCorrAttrsForSearch(osAccountInst));
-                            break;
+                } else {
+                    if (artifactTypeID == ARTIFACT_TYPE.TSK_KEYWORD_HIT.getTypeID()) {
+                        BlackboardAttribute setNameAttr = analysisResult.getAttribute(BlackboardAttribute.Type.TSK_SET_NAME);
+                        if (setNameAttr != null && CorrelationAttributeUtil.getEmailAddressAttrDisplayName().equals(setNameAttr.getValueString())) {
+                            correlationAttrs.addAll(makeCorrAttrFromArtifactAttr(analysisResult, BlackboardAttribute.ATTRIBUTE_TYPE.TSK_KEYWORD, CorrelationAttributeInstance.EMAIL_TYPE_ID, analysisResult.getAttributes()));
+                        }
+                    }
+                    Content parent = analysisResult.getParent();
+                    if (parent instanceof AbstractFile) {
+                        correlationAttrs.addAll(CorrelationAttributeUtil.makeCorrAttrsForSearch((AbstractFile) parent));
+                    } else if (parent instanceof AnalysisResult) {
+                        correlationAttrs.addAll(CorrelationAttributeUtil.makeCorrAttrsForSearch((AnalysisResult) parent));
+                    } else if (parent instanceof DataArtifact) {
+                        correlationAttrs.addAll(CorrelationAttributeUtil.makeCorrAttrsForSearch((DataArtifact) parent));
+                    } else if (parent instanceof OsAccount) {
+                        for (OsAccountInstance osAccountInst : ((OsAccount) parent).getOsAccountInstances()) {
+                            if (osAccountInst.getDataSource().equals(analysisResult.getDataSource())) {
+                                correlationAttrs.addAll(CorrelationAttributeUtil.makeCorrAttrsForSearch(osAccountInst));
+                                break;
```

Comment:
Let's comment this, something like:
"Since we search based on type and value, we only need make correlation attributes for one of the OS account instances."